### PR TITLE
ci: implement commit message validation

### PR DESCRIPTION
**Pull request by dignifiedquire**

### CODE REVIEW

**Comment by dignifiedquire** [comment]
> The package `validate-commit-msg` seems to have been deprecated in favor of `CommitLint`.
> 
> [npmjs.com/package/validate-commit-msg](https://www.npmjs.com/package/validate-commit-msg)

yep, but this script works, so I am not gonna spend time figuring out a new package

**Comment by dignifiedquire** [comment]
> @dignifiedquire - What is this PR? What motivates its existence? Why in the world does it require Node.js?

We are struggeling to keep our commit history clean and matching our commit convention that we agreed upon, so I added a ci check to help us with that, as we need to get up our game in communicating changes.

**Comment by dignifiedquire** [comment]
> Why in the world does it require Node.js?

Because it is written in javascript

File: package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/filecoin-project/rust-fil-proofs.git"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
```
[the remark this answers is not in the document]
Comment (dignifiedquire) [reply package.json]: fixed

**Comment by dignifiedquire** [comment]
> @dignifiedquire - This PR needs a `.vcmrc` file. The [default options](https://www.npmjs.com/package/validate-commit-msg#options) which are used when we don't have a `.vcmrc` file include types (`chore` and `revert`) which we do not allow as [per our CONTRIBUTING.md guide](https://github.com/filecoin-project/rust-fil-proofs/blob/master/CONTRIBUTING.md#scope) [](/filecoin-project/rust-fil-proofs/blob/HEAD@{2019-08-27T16:27:50Z}/CONTRIBUTING.md#scope)

fixed

**Comment by schomatis** [comment]
@dignifiedquire Is this ready to merge?

**Comment by dignifiedquire** [comment]
> @dignifiedquire Is this ready to merge?

yeah, I just keep updating it because of so many merges

**Pull request merged**